### PR TITLE
Toast: Here's to us

### PR DESCRIPTION
**Pull request by codecaaron**
## Toast and Toaster

- Adds Toast and Toaster to gamut templates.
- Adds framer-motion to peer dependencies.

### CODE REVIEW

File: packages/gamut-templates/src/Toasts/styles/Toast.module.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+.closeButton {
+  position: absolute;
+  right: 0;
+  top: 0;
+  padding: px-rem(5);
+  background: transparent;
+  border: none;
+  outline: none;
```
Comment (JoshuaKGoldberg) [inline packages/gamut-templates/src/Toasts/styles/Toast.module.scss]: This needs to have a visual indication of being focused. Can you use the default `Button` styles with some dark-ish theme color instead?

File: packages/styleguide/stories/Toast.stories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Toast } from '../../gamut-templates/src';
+import { AlertIcon } from '../../gamut/src';
+
+export default {
+  component: Toast,
+  title: 'Templates/Toast',
```
[the remark this answers is not in the document]
Comment (codecaaron) [reply packages/styleguide/stories/Toast.stories.js]: Went with template and it is currently living in templates as well.

File: packages/gamut-templates/src/Toasts/Toast.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import s from './styles/Toast.module.scss';
+
+type ToastProps = {
+  icon?: React.ReactElement;
+  onClick?: () => void;
+  onClose: () => void;
+} & ContainerElementProps;
+
+const Toast: React.FC<ToastProps> = ({
```
Comment (hmafzal) [inline packages/gamut-templates/src/Toasts/Toast.tsx]: We should add some tests for this component

File: packages/gamut-templates/src/Toasts/Toast.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
```
Comment (hmafzal) [inline packages/gamut-templates/src/Toasts/Toast.tsx]: Some screenshots in the description would be good! 

Separately, do we have any defined requirements for PRs that add new components to gamut? If not, I think that should be something we discuss next gamut sync

File: packages/gamut-templates/src/Toasts/Toaster.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+type ToastShape = {
+  id: string;
+  icon: any;
+  onClose: () => void;
+  onClick: () => void;
+  message: any;
+};
+
+type ToasterProps = {
+  toasts: ToastShape[];
```
Comment (hmafzal) [inline packages/gamut-templates/src/Toasts/Toaster.tsx]: We should be able to just use ToastProps here instead of defining a new shape

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.2.31-alpha.c91450.10+c9145049<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.2.31-alpha.33643f.11+33643f2e<br/></pre>

File: packages/gamut-templates/src/Toasts/Toaster.tsx (hunk 5 above)
Comment (codecaaron) [reply packages/gamut-templates/src/Toasts/Toaster.tsx]: They are different shapes, similiar, but necessarily different.

File: packages/gamut-templates/package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@codecademy/gamut-templates",
   "description": "Templates library for Gamut",
-  "version": "2.2.30",
+  "version": "2.2.28",
```
Comment (JoshuaKGoldberg) [inline packages/gamut-templates/package.json]: Merging is fun 😄

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.2.31-alpha.c176eb.12+c176eb7e<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-templates@2.2.31-alpha.1c69d5.13+1c69d50b<br/></pre>

**Pull request closed without merging**